### PR TITLE
[Issue 548] Fixed all find bugs in all modules

### PR DESCRIPTION
**Pull request by ravipesala**
This PR fixes all find bug issues in all modules

### CODE REVIEW

File: hadoop/src/main/java/org/carbondata/hadoop/util/ObjectSerializationUtil.java [hunk 1]
```diff
@@ -33,18 +37,29 @@ public static String convertObjectToString(Object obj) throws IOException {
...
+          oos.close();
+        }
+        if (gos != null) {
+          gos.close();
+        }
+        if (baos != null) {
+          baos.close();
+        }
+      } catch (IOException e) {
+        LOG.error(e);
```
Comment (jackylk) [inline hadoop/src/main/java/org/carbondata/hadoop/util/ObjectSerializationUtil.java]: should throw the exception again, right?

Comment (ravipesala) [inline hadoop/src/main/java/org/carbondata/hadoop/util/ObjectSerializationUtil.java]: Throwing the exception in finally block is not recommendable as it confuses the user and masks the actual exceptions. In two cases the flow comes to finally block
 First case is after successful completion of flow in try block and reaches finally block , in this case even though the exception the thrown while closing the connections cannot effect the flow, so we just log it.
Second case is something abnormal happened in try block flow and throws exception, so before throwing exception it goes to finally block to close the connections. In this case if we throw exception from finally block , it masks the actual exception.  That is why throwing exception in finally block is not recommended.

**Comment by gvramana** [comment]
@ravipesala Please rebase

**Comment by ravipesala** [comment]
@gvramana Rebased. Please review and merge

**Comment by jackylk** [comment]
LGTM

**Pull request merged**